### PR TITLE
fix: badly initialized atoms.

### PR DESCRIPTION
**Pull request by Evankhell561**
:cl:
code: fix some bug reports from the atom subsystem.
/:cl:

### CODE REVIEW

**Review by genessee596 — CHANGES_REQUESTED** [review]
make changelog plox

good for players to see we're actually fixing buges

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -1,5 +1,4 @@
 /mob/Destroy()//This makes sure that mobs with clients/keys are not just deleted from the game.
-	STOP_PROCESSING(SSmobs, src)
```
Comment (TorinoFermic) [inline code/modules/mob/mob.dm:2]: I would like to know why this line got removed

File: code/game/machinery/kitchen/microwave.dm [hunk 2]
```diff
@@ -124,7 +123,7 @@
 			to_chat(user, SPAN_WARNING("It's dirty!"))
 			return 1
 	else if(is_type_in_list(I,acceptable_items))
-		if (contents.len>=max_n_of_items)
+		if(contents.len>=max_n_of_items)
```
Comment (AlexMorgan3817) [inline code/game/machinery/kitchen/microwave.dm:126]:
```suggestion
		if(length(contents) >= max_n_of_items)
```

File: code/game/machinery/kitchen/microwave.dm [hunk 3]
```diff
@@ -147,10 +146,10 @@
 	        istype(I,/obj/item/weapon/reagent_containers/food/drinks) || \
 	        istype(I,/obj/item/weapon/reagent_containers/food/condiment) \
 		)
-		if (!I.reagents)
+		if(!I.reagents)
 			return 1
 		for (var/datum/reagent/R in I.reagents.reagent_list)
-			if (!(R.id in acceptable_reagents))
+			if(!(R.id in acceptable_reagents))
```
Comment (AlexMorgan3817) [inline code/game/machinery/kitchen/microwave.dm:152]:
```suggestion
			if(!acceptable_reagents.Find(R.id))
```

File: code/game/machinery/kitchen/microwave.dm [hunk 4]
```diff
@@ -201,42 +200,42 @@
...
 				items_measures_p[display_name] = "turnovers"
-			if (istype(O,/obj/item/weapon/reagent_containers/food/snacks/meat/carp))
+			if(istype(O,/obj/item/weapon/reagent_containers/food/snacks/meat/carp))
 				items_measures[display_name] = "fillet of meat"
 				items_measures_p[display_name] = "fillets of meat"
 			items_counts[display_name]++
 		for (var/O in items_counts)
 			var/N = items_counts[O]
-			if (!(O in items_measures))
+			if(!(O in items_measures))
```
Comment (AlexMorgan3817) [inline code/game/machinery/kitchen/microwave.dm:222]:
```suggestion
			if(!items_measures.Find(O))
```

File: code/game/machinery/kitchen/microwave.dm [hunk 5]
```diff
@@ -201,42 +200,42 @@
...
 				dat += {"<B>[capitalize(O)]:</B> [N] [lowertext(O)]\s<BR>"}
 			else
-				if (N==1)
+				if(N==1)
```
Comment (AlexMorgan3817) [inline code/game/machinery/kitchen/microwave.dm:225]:
```suggestion
				if(N == 1)
```

File: code/game/objects/items/weapons/melee/energy.dm [hunk 6]
```diff
@@ -195,7 +197,7 @@
 	var/cleanup = TRUE	// Should the blade despawn moments after being discarded by the summoner?
 
 /obj/item/weapon/melee/energy/blade/New()
-
+	..()
```
Comment (AlexMorgan3817) [inline code/game/objects/items/weapons/melee/energy.dm:200]:
```suggestion
	. = ..()
```

File: code/game/objects/structures/cyberplants.dm [hunk 7]
```diff
@@ -31,7 +31,7 @@
 	var/sanity_value = 0.2
 
 /obj/structure/cyberplant/Initialize()
-	..()
+	.=..()
```
Comment (AlexMorgan3817) [inline code/game/objects/structures/cyberplants.dm:34]:
```suggestion
	. = ..()
```

File: code/modules/admin/bluespacetech.dm [hunk 8]
```diff
@@ -295,8 +295,9 @@ ADMIN_VERB_ADD(/client/proc/cmd_dev_bst, R_ADMIN|R_DEBUG, TRUE)
 	desc = "An ID straight from Central Command. This one looks highly classified."
 	spawn_frequency = 0
 
-/obj/item/weapon/card/id/bst/New()
-		access = get_all_accesses()+get_all_centcom_access()+get_all_syndicate_access()
+/obj/item/weapon/card/id/bst/Initialize()
+	. = ..()
+	access = get_all_accesses()+get_all_centcom_access()+get_all_syndicate_access()
```
Comment (AlexMorgan3817) [inline code/modules/admin/bluespacetech.dm:300]:
```suggestion
	access = get_all_accesses() + get_all_centcom_access() + get_all_syndicate_access()
```

File: code/modules/hydroponics/seed_packets.dm [hunk 9]
```diff
@@ -74,6 +74,7 @@ var/global/list/plant_seed_sprites = list()
 	seed_type = null
 
 /obj/item/seeds/random/New()
+	..()
```
Comment (AlexMorgan3817) [inline code/modules/hydroponics/seed_packets.dm:77]:
```suggestion
	. = ..()
```

File: code/modules/mining/coins.dm [hunk 10]
```diff
@@ -15,6 +15,7 @@
 /obj/item/weapon/coin/New()
 	pixel_x = rand(0,16)-8
 	pixel_y = rand(0,8)-8
+	..()
```
Comment (AlexMorgan3817) [inline code/modules/mining/coins.dm:18]:
```suggestion
	. = ..()
```

File: code/modules/mining/mine_turfs.dm [hunk 11]
```diff
@@ -36,7 +36,7 @@
 	has_resources = 1
 
 /turf/simulated/mineral/Initialize()
-	..()
+	.=..()
```
Comment (AlexMorgan3817) [inline code/modules/mining/mine_turfs.dm:39]:
```suggestion
	. = ..()
```

File: code/modules/mining/ore.dm [hunk 12]
```diff
@@ -101,6 +101,7 @@
 /obj/item/weapon/ore/New()
 	pixel_x = rand(0,16)-8
 	pixel_y = rand(0,8)-8
+	..()
```
Comment (AlexMorgan3817) [inline code/modules/mining/ore.dm:104]:
```suggestion
	. = ..()
```

File: code/modules/mob/living/bot/drillbot.dm [hunk 13]
```diff
@@ -22,6 +22,7 @@
 
 /mob/living/bot/miningonestar/Destroy()
 	loot = null
+	.=..()
```
Comment (AlexMorgan3817) [inline code/modules/mob/living/bot/drillbot.dm:25]:
```suggestion
	. = ..()
```

File: code/modules/mob/living/bot/drillbot.dm [hunk 14]
```diff
@@ -86,4 +87,4 @@
 		if(get_dist(src, AST) >= 1)
 			UnarmedAttack(AST)
 		else
-			walk_to(src,AST,1,5,0)
\ No newline at end of file
+			walk_to(src,AST,1,5,0)
```
Comment (AlexMorgan3817) [inline code/modules/mob/living/bot/drillbot.dm:90]:
```suggestion
			walk_to(src, AST, 1, 5, 0)
```

File: code/modules/mob/living/silicon/decoy/decoy.dm [hunk 15]
```diff
@@ -5,11 +5,9 @@
 	anchored = TRUE // -- TLE
 	movement_handlers = list(/datum/movement_handler/no_move)
 
-/mob/living/silicon/decoy/New()
+/mob/living/silicon/decoy/Initialize()
 	src.icon = 'icons/mob/AI.dmi'
 	src.icon_state = "ai"
 	src.anchored = TRUE
 	src.canmove = 0
```
Comment (AlexMorgan3817) [inline code/modules/mob/living/silicon/decoy/decoy.dm:12]:
```suggestion
	icon = 'icons/mob/AI.dmi'
	icon_state = "ai"
	anchored = TRUE
	canmove = 0
```

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 16]
```diff
@@ -101,12 +101,13 @@ var/global/list/robot_modules = list(
 
 
 /obj/item/weapon/robot_module/Initialize()
+	.=..()
```
Comment (AlexMorgan3817) [inline code/modules/mob/living/silicon/robot/robot_modules.dm:104]:
```suggestion
	. = ..()
```

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 17]
```diff
@@ -116,13 +116,13 @@
 
 	verbs -= /mob/verb/observe
 
-	if (mob_size)
+	if(mob_size)
 		nutrition_step = mob_size * 0.03 * metabolic_factor
 		bite_factor = mob_size * 0.1
 		max_nutrition *= 1 + (nutrition_step*4)//Max nutrition scales faster than costs, so bigger creatures eat less often
-		reagents = new/datum/reagents(stomach_size_mult*mob_size, src)
+		create_reagents(stomach_size_mult*mob_size)
```
Comment (AlexMorgan3817) [inline code/modules/mob/living/simple_animal/simple_animal.dm:123]:
```suggestion
		create_reagents(stomach_size_mult * mob_size)
```

File: code/modules/organs/internal/carrion.dm [hunk 18]
```diff
@@ -399,7 +399,7 @@
 	anchored = TRUE
 
 /obj/effect/decal/cleanable/carrion_puddle/Initialize()
-	..()
+	.=..()
```
Comment (AlexMorgan3817) [inline code/modules/organs/internal/carrion.dm:402]:
```suggestion
	. = ..()
```

File: code/modules/overmap/spacetravel.dm [hunk 19]
```diff
@@ -19,6 +19,7 @@ var/list/cached_space = list()
 /obj/effect/overmap/sector/temporary/Destroy()
 	map_sectors["[map_z]"] = null
 	testing("Temporary sector at [x],[y] was deleted.")
+	.=..()
```
Comment (AlexMorgan3817) [inline code/modules/overmap/spacetravel.dm:22]:
```suggestion
	. = ..()
```

File: code/modules/paperwork/photography.dm [hunk 20]
```diff
@@ -44,6 +44,7 @@ var/global/photo_count = 0
 
 /obj/item/weapon/photo/New()
 	id = photo_count++
+	..()
```
Comment (AlexMorgan3817) [inline code/modules/paperwork/photography.dm:47]:
```suggestion
	. = ..()
```

File: code/modules/projectiles/guns/projectile/dartgun.dm [hunk 21]
```diff
@@ -4,14 +4,13 @@
...
 	kill_count = 15 //shorter range
-
 	muzzle_type = null
+	var/reagent_amount = 15
 
 /obj/item/projectile/bullet/chemdart/New()
-	reagents = new/datum/reagents(reagent_amount)
-	reagents.my_atom = src
+	create_reagents(reagent_amount)
+	..()
```
Comment (AlexMorgan3817) [inline code/modules/projectiles/guns/projectile/dartgun.dm:13]:
```suggestion
	. = ..()
```

File: code/modules/projectiles/guns/projectile/dartgun.dm [hunk 22]
```diff
@@ -191,4 +190,4 @@
 	else if (href_list["eject_cart"])
 		unload_ammo(usr)
 	src.updateUsrDialog()
-	return
\ No newline at end of file
+	return
```
Comment (AlexMorgan3817) [inline code/modules/projectiles/guns/projectile/dartgun.dm:193]:
```suggestion

```

File: code/modules/projectiles/guns/projectile/dartgun.dm [hunk 23]
```diff
@@ -191,4 +190,4 @@
 	else if (href_list["eject_cart"])
 		unload_ammo(usr)
 	src.updateUsrDialog()
```
Comment (AlexMorgan3817) [inline code/modules/projectiles/guns/projectile/dartgun.dm:192]:
```suggestion
	updateUsrDialog()
```

File: code/modules/research/xenoarchaeology/finds/finds_fossils.dm [hunk 24]
```diff
@@ -59,6 +60,7 @@
 /obj/skeleton/New()
 	src.breq = rand(6)+3
 	src.desc = "An incomplete skeleton, looks like it could use [src.breq-src.bnum] more bones."
+	..()
```
Comment (AlexMorgan3817) [inline code/modules/research/xenoarchaeology/finds/finds_fossils.dm:63]:
```suggestion
	. = ..()
```

File: code/modules/research/xenoarchaeology/finds/finds_fossils.dm [hunk 25]
```diff
@@ -12,15 +12,16 @@
...
+	. = ..()
 	var/list/l = list("/obj/item/weapon/fossil/bone"=9,"/obj/item/weapon/fossil/skull"=3,
 	"/obj/item/weapon/fossil/skull/horned"=2)
 	var/t = pickweight(l)
 	var/obj/item/weapon/W = new t(src.loc)
 	var/turf/T = get_turf(src)
 	if(istype(T, /turf/simulated/mineral))
 		T:last_find = W
-	qdel(src)
+	return INITIALIZE_HINT_QDEL
```
Comment (AlexMorgan3817) [inline code/modules/research/xenoarchaeology/finds/finds_fossils.dm:24]: Are you sure?

File: code/modules/research/xenoarchaeology/finds/finds_fossils.dm [hunk 26]
```diff
@@ -104,4 +106,5 @@
 	animal = 0
 
 /obj/item/weapon/fossil/plant/New()
+	..()
```
Comment (AlexMorgan3817) [inline code/modules/research/xenoarchaeology/finds/finds_fossils.dm:109]:
```suggestion
	. = ..()
```

File: code/modules/research/xenoarchaeology/geosample.dm [hunk 27]
```diff
@@ -29,8 +30,7 @@
 	icon_state = "sliver[rand(1,3)]"
 	pixel_x = rand(0,16)-8
 	pixel_y = rand(0,8)-8
-	create_reagents(50)
-	reagents.add_reagent("ground_rock",50)
+	..()
```
Comment (AlexMorgan3817) [inline code/modules/research/xenoarchaeology/geosample.dm:33]:
```suggestion
	. = ..()
```

File: code/modules/research/xenoarchaeology/machinery/coolant.dm [hunk 28]
```diff
@@ -6,7 +6,7 @@
 	amount_per_transfer_from_this = 10
 
 /obj/structure/reagent_dispensers/coolanttank/Initialize(mapload, ...)
-	..()
+	.=..()
```
Comment (AlexMorgan3817) [inline code/modules/research/xenoarchaeology/machinery/coolant.dm:9]:
```suggestion
	. = ..()
```

File: code/modules/research/xenoarchaeology/tools/ano_device_battery.dm [hunk 29]
```diff
@@ -9,6 +9,7 @@
 	var/effect_id = ""
 
 /obj/item/weapon/anobattery/New()
+	..()
```
Comment (AlexMorgan3817) [inline code/modules/research/xenoarchaeology/tools/ano_device_battery.dm:12]:
```suggestion
	. = ..()
```

File: code/modules/virus2/items_devices.dm [hunk 30]
```diff
@@ -108,6 +108,7 @@
 	var/analysed = 1
 
 /obj/item/weapon/diseasedisk/premade/New()
+	..()
```
Comment (AlexMorgan3817) [inline code/modules/virus2/items_devices.dm:111]:
```suggestion
	. = ..()
```

File: code/modules/stashes/stash_spawner.dm [hunk 31]
```diff
@@ -11,17 +11,18 @@
...
 		return INITIALIZE_HINT_QDEL
 
 	datum.select_location()
-	if (!datum.stash_location)
+	if(!datum.stash_location)
 		//We require a found location
 		return INITIALIZE_HINT_QDEL
 
-	if (!isStationLevel(src.z))
+	if(!isStationLevel(src.z))
```
Comment (AlexMorgan3817) [inline code/modules/stashes/stash_spawner.dm:25]:
```suggestion
	if(!isStationLevel(z))
```

File: code/modules/stashes/stash_spawner.dm [hunk 32]
```diff
@@ -11,17 +11,18 @@
 
 
 /obj/item/stash_spawner/Initialize()
+	.=..()
```
Comment (AlexMorgan3817) [inline code/modules/stashes/stash_spawner.dm:14]:
```suggestion
	. = ..()
```

**Review by AlexMorgan3817 — CHANGES_REQUESTED** [review]

**Review by Hopekz — APPROVED** [review]

**Comment by TorinoFermic** [comment]
I would like have this PR merged as soon all suggested changes are committed to this PR.

**Review by genessee596 — APPROVED** [review]
thank you

**Comment by Evankhell561** [comment]
> I would like have this PR merged as soon all suggested changes are committed to this PR.

the suggestions that I have not resolved yet, do not have to do with my pr. I don't have the time or the adequate internet to take charge.

**Comment by Hopekz** [comment]
All you have to do is check the "allow maintainers to edit" box you unchecked before and we'll be able to help out.

**Comment by Hopekz** [comment]
> I don't have the time or the adequate internet to take charge.

All you have to do is check the "allow maintainers to edit" box you unchecked before and we'll be able to help out.

**Comment by Evankhell561** [comment]
> > I don't have the time or the adequate internet to take charge.
> 
> All you have to do is check the "allow maintainers to edit" box you unchecked before and we'll be able to help out.


I don't remember disabling that option.

**Comment by Hopekz** [comment]
> I don't remember disabling that option.

Okay, well, not sure what to say except that it is currently unchecked, preventing us from helping out.

**Comment by Evankhell561** [comment]
ready

**Review by TorinoFermic — APPROVED** [review]

**Review by Hopekz — APPROVED** [review]

**Pull request merged**